inspire_utils/name.py

```diff
@@ -200,14 +200,22 @@ def _is_roman_numeral(suffix):
         return final_name
 
     @classmethod
-    def from_parts(cls, first, last, middle=None, suffix=None, title=None):
-        name = HumanName()
+    def from_parts(
+        cls,
+        first=None,
+        last=None,
+        middle=None,
+        suffix=None,
+        title=None,
+        constants=None,
+    ):
+        name = HumanName(constants=constants)
```

Comment:
I did not mention it explicitly previously, but most importantly, you should handle the default constants as in https://github.com/inspirehep/inspire-utils/pull/38/files/79e4828c90715279e6b02f7a5683198897779f94..a83da4632ee79e3c9c0229bba50ee086660434f1#diff-f116d82c04785f5d37d57c86cae932cdR75.

Alternatively, maybe it's simpler to do `name = ParsedName()`, assign to `name._parsed_name.first`, etc., and `capitalize` at the end like the `__init__`. Then you don't have to handle constants or to modify the `init` to allow taking a `HumanName` instance.

Reply:
As discussed IRL, we don't need `constants` in this case, as the meaning of the parts is already explicitly specified.